### ISSUE TITLE
Default cache value for SEQUENCE not consistent between explicitly vs implicitly created

### ISSUE THREAD

**Issue by ashwinstar** [greenplum-db/gpdb#14851]
### Greenplum version or build
7X

### Expected behavior
Default Cache value should be consistent between manually created sequence and sequence created by SERIAL or GENERATED BY DEFAULT AS IDENTITY.

### Actual behavior
Currently only for manual created sequence the default value is 20. For sequence created by SERIAL or GENERATED BY 
DEFAULT AS IDENTITY, cache value is still 1

```sql
demo=# create sequence t_seq2;
CREATE SEQUENCE
demo=# \d+ t_seq2      
                           Sequence "public.t_seq2"
  Type  | Start | Minimum |       Maximum       | Increment | Cycles? | Cache 
--------+-------+---------+---------------------+-----------+---------+-------
 bigint |     1 |       1 | 9223372036854775807 |         1 | no      |    20
```
But it is NOT true when the sequence is created by SERIAL or GENERATED BY DEFAULT AS IDENTITY

```sql
demo=# create table t_seq (id serial, b text);
CREATE TABLE
demo=# \d+ t_seq
                                                Table "public.t_seq"
 Column |  Type   | Collation | Nullable |              Default              | Storage  | Stats target | Description 
--------+---------+-----------+----------+-----------------------------------+----------+--------------+-------------
 id     | integer |           | not null | nextval('t_seq_id_seq'::regclass) | plain    |              | 
 b      | text    |           |          |                                   | extended |              | 
Distributed by: (id)
Access method: heap

demo=# \d+ t_seq_id_seq
                    Sequence "public.t_seq_id_seq"
  Type   | Start | Minimum |  Maximum   | Increment | Cycles? | Cache 
---------+-------+---------+------------+-----------+---------+-------
 integer |     1 |       1 | 2147483647 |         1 | no      |     1
Owned by: public.t_seq.id
```

### Step to reproduce the behavior
Provided above.

**Comment by hyongtao-db** [comment]
I do some spike work about this issue.

Let's focus on these two files, src/backend/parser/parser_utilcmd.c and src/backend/commands/sequence.c.

Please refer to line 561-568 in src/backend/parser/parser_utilcmd.c
```
	/*
	 * gpdb sequence default cache is 20 to avoid frequent sequence value apply
	 * in QE, we do not need this optimize here. Since each input data populate
	 * serial column in QD and then dispatch to QE
	 */
	if (!has_cache_option)
		seqstmt->options = lappend(seqstmt->options,
								   makeDefElem("cache", (Node *) makeInteger((long) 1), -1));
```
and line 1795-1803 in src/backend/commands/sequence.c

```
	else if (isInit)
	{
		/*
		 * PostgreSQL default value is 1, GPDB privately bump up to 20.
		 * If a sequence in UDF, QE executor need to apply sequence value from QD.
		 * Frequent sequence application is network bottleneck for query execution.
		 */
		seqform->seqcache = 20;
	}
```

If we rudely change `makeDefElem("cache", (Node *) makeInteger((long) 1), -1))` to `makeDefElem("cache", (Node *) makeInteger((long) 20), -1))`, we will get these error:


```
gpadmin=# CREATE TABLE table_14851(f text,id serial);
CREATE TABLE
gpadmin=# \d+ table_14851_id_seq
                 Sequence "public.table_14851_id_seq"
  Type   | Start | Minimum |  Maximum   | Increment | Cycles? | Cache
---------+-------+---------+------------+-----------+---------+-------
 integer |     1 |       1 | 2147483647 |         1 | no      |    20
Owned by: public.table_14851.id

gpadmin=# INSERT INTO table_14851 (f) VALUES ('test_defaults');
INSERT 0 1
gpadmin=# SELECT * FROM table_14851;
       f       | id
---------------+----
 test_defaults |  1
(1 row)

gpadmin=# SELECT nextval('table_14851_id_seq');
 nextval
---------
      21
(1 row)
```

It's a wrong behavior, The two sets of values (_insert into table_ and _nextval_) have lost continuity.

**Comment by yanwr1** [comment]
the current behavior for sequence in gpdb7:
In Greenplum we manage dispatcher and executor sequence ranges separately. In qe, firstly get the nextval value from the cached, after exhaust the cached numbers then it will forward a nextval request to the QD.

And in qd, there are two ways to call nextval: request from customer and qe, they will invoke 'nextval_internal' with called_from_dispatcher=false and called_from_dispatcher=true. So in a hash table, there are two entry for a sequence, which saves different sequence data.

When the nextval request is from customer, it will firstly get the nextval value from the cache. otherwise it will read the latest sequence data from log and forward cache numbers as the next value. If the nextval request is from qe, it will pass the first step and forward directly.

```
create sequence seq_t1;
\d+ seq_t1
                           Sequence "public.seq_t1"
  Type  | Start | Minimum |       Maximum       | Increment | Cycles? | Cache 
--------+-------+---------+---------------------+-----------+---------+-------
 bigint |     1 |       1 | 9223372036854775807 |         1 | no      |    20

create table t1(a int, b int default nextval('seq_t1') not null);

select nextval('seq_t1');
 nextval 
---------
      61
(1 row)
``` 
create a same sequence, and execute 'select nextval' before insert statement:
```
postgres=# create  sequence seq_t1;
CREATE SEQUENCE
postgres=# select nextval('seq_t1');
 nextval 
---------
       1
(1 row)

postgres=# create table t1(a int, b int default nextval('seq_t1') not null);

CREATE TABLE
postgres=# insert into t1(a) select * from generate_series(1,10);
INSERT 0 10
postgres=# select nextval('seq_t1');
 nextval 
---------
       2
(1 row)
```
If we change the serial's cache to 20, they have the same performance.

**Comment by kainwen** [comment]
@yanwr1 @adam8157 what is this issue's status? Thanks!

**Comment by adam8157** [comment]
Fixed by #15204, and this sequence cache has a known issue https://github.com/greenplum-db/gpdb/issues/15837 which we won't fix for now.
